api/v1beta1/kafkacluster_types.go

```diff
@@ -273,6 +274,10 @@ type EnvoyConfig struct {
 	HealthCheckPort *int32 `json:"healthCheckPort,omitempty"`
 	// DisruptionBudget is the pod disruption budget attached to Envoy Deployment(s)
 	DisruptionBudget *DisruptionBudgetWithStrategy `json:"disruptionBudget,omitempty"`
+	// key-value map of additional envoy configuration parameters
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +optional
+	AdditionalConfig map[string]string `json:"additionalConfig,omitempty"`
```

Comment:
I opted to add a free form `additionalConfig` bag for additional configurations as we have other envoy specific configurations we may add here. let me know if this sounds ok with you

Reply:
Should this be rather called `Args` or `CommandLineArgs` to separate the configuration that is passed in via the config map from config passed in via command line args?

We could still have  `AddtionalConfig` for those additional configs which would go into the configmap.

Reply:
that's a good suggestion.
One thing though I am not sure about is if we should simply add all k-v pairs as arguments to envoy command line:
https://github.com/banzaicloud/koperator/pull/714/files#diff-821e43efb03a091583c0af4c4906d28f7186b6c7909d8f9dbbe0120f814635d4R69

or still handle them individually like concurrency.
not all https://www.envoyproxy.io/docs/envoy/latest/operations/cli#command-line-options may be valid when running in the operator

Reply:
I would use this for all cli options except `-c`, `--config-path` which is set by koperator in order to be able to pass in the path to the config file it generated.